src/Mvc/Mvc.Core/src/ModelBinding/Binders/DateTimeModelBinder.cs

```diff
@@ -0,0 +1,107 @@
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
+{
+    /// <summary>
+    /// An <see cref="IModelBinder"/> for <see cref="DateTime"/> and nullable <see cref="DateTime"/> models.
+    /// </summary>
+    public class DateTimeModelBinder : IModelBinder
+    {
+        private readonly DateTimeStyles _supportedStyles;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DateTimeModelBinder"/>.
+        /// </summary>
+        /// <param name="supportedStyles">The <see cref="DateTimeStyles"/>.</param>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
+        public DateTimeModelBinder(DateTimeStyles supportedStyles, ILoggerFactory loggerFactory)
+        {
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _supportedStyles = supportedStyles;
+            _logger = loggerFactory.CreateLogger<DateTimeModelBinder>();
+        }
+
+        /// <inheritdoc />
+        public Task BindModelAsync(ModelBindingContext bindingContext)
+        {
+            if (bindingContext == null)
+            {
+                throw new ArgumentNullException(nameof(bindingContext));
+            }
+
+            _logger.AttemptingToBindModel(bindingContext);
+
+            var modelName = bindingContext.ModelName;
+            var valueProviderResult = bindingContext.ValueProvider.GetValue(modelName);
+            if (valueProviderResult == ValueProviderResult.None)
+            {
+                _logger.FoundNoValueInRequest(bindingContext);
+
+                // no entry
+                _logger.DoneAttemptingToBindModel(bindingContext);
+                return Task.CompletedTask;
+            }
+
+            var modelState = bindingContext.ModelState;
+            modelState.SetModelValue(modelName, valueProviderResult);
+
+            var metadata = bindingContext.ModelMetadata;
+            var type = metadata.UnderlyingOrModelType;
+            try
+            {
+
+                var value = valueProviderResult.FirstValue;
+
+                object model;
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    // Parse() method trims the value (with common DateTimeSyles) then throws if the result is empty.
+                    model = null;
+                }
+                else if (type == typeof(DateTime))
+                {
+                    model = DateTime.Parse(value, valueProviderResult.Culture, _supportedStyles);
+                }
+                else
+                {
+                    // unreachable
+                    throw new NotSupportedException();
```

Comment:
This type is constructable right? So it isn't really unreachable.

Reply:
Maybe because this uncommon? IDK, removed it because it didn't really add much value.